build/localnet/manifests/cli-client.yaml

```diff
@@ -76,6 +76,8 @@ spec:
               # Any host that is visible and connected to the cluster can be arbitrarily selected as the RPC host
             - name: RPC_HOST
               value: v1-validator001
+            - name: POCKET_P2P_HOSTNAME
+              value: "127.0.0.1"
```

Comment:
I added this to resolve this error when using the debug cli in the tilt localnet:
```
{"level":"fatal","error":"parsing multiaddr from config: resolving peer IP for hostname: : resolving peer IP for hostname: , lookup : no such host","time":"2023-04-13T07:21:23Z","message":"Failed to create p2p module"}
command terminated with exit code 1
```

I read this as, it tried to resolve DNS for an empty string

I hypothesize that the reason this is necessary in tilt  and not in docker compose is because the config manifest  has `p2p.is_empty_connection_type: false` and is common to both the validator and the debug cli manifests. I'm attempting to set `POCKET_P2P_IS_CONNECTION_EMPTY` but it didn't seem to work straight away. :thinking: 

```yaml
  - name: POCKET_P2P_IS_EMPTY_CONNECTION_TYPE
    value: "true"
```

Reply:
@bryanchriswhite I think you don't see that behavior on docker-compose because by default CLI uses `client1.json` as a main config: https://github.com/pokt-network/pocket/blob/39d16cd7adaf03dfd329f06a8935399a9ec8b068/app/client/cli/debug.go#L80
Which has `node1.consensus` set as a hostname.

I need to look deeper to figure out why it tries to resolve the empty host string when the value is `127.0.0.1`, I wonder what'd happen if the value was `localhost`. I'd vote for `0.0.0.0` as a default value.

I think requiring such parameters (hostname, Postgres, etc.) is counterintuitive when the user only needs to use CLI. In the current v0 world, users often use CLI to query pocket RPC endpoints to get the latest height, information about nodes, etc. We definitely should not require unnecessary configs. I'll see if such a ticket exists to add more information or create a new one tomorrow.

Reply:
@okdas Apologies for any confusion, to be clear, setting  `POCKET_P2P_HOSTNAME` to `127.0.0.1` **does resolve the error**, I just wasn't confident that it was the correct or even temporarily acceptable solution. Given your improved explanation for the behavior, it seems like a change need to be made in the code rather than the configs. It shouldn't be necessary for the debug client to specify a hostname that it doesn't use (see #678).

Reply:
> 1. Isn't `is_empty_connection_type` only used for testing? The CLI still needs a non-empty connection type (i.e. TLS) to communicate with the network.

`runtime/configs.P2PConfig#ConnectionType` is only used to inform the listen address that is provided to the libp2p host. In the debug client case, we use `libp2p.NoListenAddrs` but at some other point, we try to construct a self service URL from the P2P config and parse it as a multiaddr anyways.

> 2. W/e solution we decide to go with, adding comments in the yaml file next to the config saying "it's a hack or needed because of Y" would help

:+1:

Reply:
👍